src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -35,10 +35,17 @@ object Lorre extends App with LazyLogging {
   lazy val db = DatabaseUtil.db
   val tezosNodeOperator = new TezosNodeOperator(TezosNodeInterface)
 
+  val numberOfRetries = conf.getInt("lorre.numberOfRetries")
+  @tailrec
+  def retry[A](attempts: Int)(fn: => Try[A]): Try[A] = fn match {
+    case Failure(_) if attempts > 1 => retry(attempts-1)(fn)
+    case noMoreAttempts => noMoreAttempts
+  }
+
   @tailrec
   def mainLoop(iteration: Int): Unit = {
-      processTezosBlocks()
-      processTezosAccounts()
+      retry(numberOfRetries)(processTezosBlocks())
```

Comment:
Why do we retry the entire operation of processing all blocks or all accounts? Why not retry at the level of retrying individual blocks or accounts? The latter would be a lot more efficient. You don't want to re-download a million accounts just because one failed.

Reply:
@vishakh what would happen if the process can't complete and only part of the blocks were downloaded?
I guess that the next iteration would retry the missing pieces?

What about failing and being unable to complete accounts? That would look even more undesirable, as we'll show a partially inconsistent situation, until the rest of the accounts would be downloaded... and that would not happen if I understand correctly that the next iteration would think that no new block was added and no accounts need downloading?

This leads to considerations about transactionality of the process operations that would need more thought, I think.

Reply:
I agree with Ivano on this one, but will be willing to discuss more in person tomorrow.